script.js

```diff
@@ -1,12 +1,225 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// Getting our structural elements from HTML
+const bodyEl = document.getElementById("bodyEl");
+const showDropDown = document.getElementById("showSelection");
+const episodeDropDown = document.getElementById("episodeSelection");
+const searchBar = document.getElementById("search");
+
+// We create some global variables here whose values can be changed later in the code
+let allShows;
+let allEpisodes;
+
+// Page setup function
+
+function pageSetUp() {
+  allShows = getAllShows();
+  // Lets Sort our shows
+  const sortedShows = allShows.sort((a, b) => {
+    return a.name < b.name ? -1 : 1;
+  });
+  createAllShowsOptions(sortedShows);
+  populateCards(sortedShows);
+}
+
+window.onload = pageSetUp;
+
+// Lets Get Live Data
+function getEpisodes(showId) {
+  let givenApi = `https://api.tvmaze.com/shows/${showId}/episodes`;
+
+  return fetch(givenApi).then(function (response) {
+    if (response.ok) {
+      return response.json();
+    }
+    throw `${response.status} ${response.statusText}`;
+  });
+}
+
+function showCount(search) {
+  let countEl = document.getElementById("count");
+  let totalEpisode = allEpisodes.length;
+  let searchedEpisode = search.length;
+  countEl.innerText = `Showing ${searchedEpisode}/ ${totalEpisode} Episodes.`;
+  return countEl;
+}
+
+function removeShowCount() {
+  let countEl = document.getElementById("count");
+  countEl.innerHTML = "";
+  return countEl;
+}
+// Writing the name and number in correct syntax
+function displayNameNumber(episode) {
+  let formattedName;
+  let episodeNum = episode.number;
+  let seasonNum = episode.season;
+
+  if (episodeNum < 10) {
+    episodeNum = `0${episode.number}`;
+  }
+  if (seasonNum < 10) {
+    seasonNum = `0${episode.season}`;
+  }
+  if (episodeNum || seasonNum) {
+    formattedName = `S${seasonNum}E${episodeNum} - ${episode.name}`;
+  } else {
+    formattedName = episode.name;
+  }
+  return formattedName;
+}
+
+// Search Bar event Listener
+searchBar.addEventListener("input", (e) => {
+  e.preventDefault();
```

Comment:
I'm not quite sure what e.preventDefault() does here? I've only ever used it for submit buttons to not refresh the page onclick

Reply:
I used to it prevent it from refreshing the page. If this is useless I can remove :)